src/context/ListContext.tsx

```diff
@@ -0,0 +1,201 @@
+import {
+  FC, createContext, useState, useEffect, ReactNode,
+} from 'react';
+import {
+  User, Post, Comment, CommentData,
+} from '../types';
+import { getUsers } from '../api/users';
+import { getPosts } from '../api/posts';
+import { deleteComment, getComments, postComment } from '../api/comments';
+
+type Props = {
+  children: ReactNode;
+};
+
+interface IGlobalContext {
+  error: boolean;
+  resetError: () => void;
+}
+
+interface IPostsContext {
+  posts: Post[];
+  isPostsLoading: boolean;
+  selectedPost: Post | null,
+  onSelectPost: (post: Post) => void;
+}
+
+interface IUsersContext {
+  users: User[];
+  selectedUser: User | null;
+  isUsersLoading: boolean;
+  onSelectUser: (user: User) => void;
+}
+
+interface ICommentsContext {
+  comments: Comment[];
+  isCommentsLoading: boolean;
+  commentIsProcessing: boolean;
+  onAddNewComment: (comment: CommentData) => Promise<boolean>;
+  onDeleteComment: (commentId: number) => void;
+}
+
+export const GlobalContext = createContext({} as IGlobalContext);
+export const PostsContext = createContext({} as IPostsContext);
```

Comment:
You are creating different contexts for each type of data (users, posts, comments). Consider using just one context for all data to simplify the structure and avoid nested context providers.